src/tools/miri/src/borrow_tracker/stacked_borrows/mod.rs

```diff
@@ -71,7 +70,7 @@ impl NewPermission {
                         access: None,
                         protector: None,
                     }
-                } else if pointee.is_unpin(*cx.tcx, cx.param_env()) {
+                } else if pointee.is_unpin(*cx.tcx, ty::ParamEnv::reveal_all()) {
```

Comment:
FWIW for cases I these I wish `is_unpin` and similar functions would accept a single argument that is `HasTyCtxt` and `HasParamEnv`/`HasTypingEnv`, so one doesn't have to pass around two things all the time...